interfaces/devdoc/sync_requirements.md

```diff
@@ -0,0 +1,42 @@
+# sync
+================
+
+## Overview
+
+`sync` provides platform-independent synchronization primitives.

```

Comment:
>primitives [](start = 53, length = 10)

I'd briefly define the two primitives we have in the overview.

 - wait : waits for a value to change from...
 - wake : wakes up one or all threads that are waiting...